misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/evaluation/engine/EvaluationTypeResolver.java

```diff
@@ -0,0 +1,186 @@
+     * @return a collection of resolved types
+     * @throws EvaluationException if unsupported type(s) found
+     */
+    public abstract List<Value> resolve(T typeDescriptor) throws EvaluationException;
+
+    /**
+     * Creates a 'BUnionType' instance by combining all member types.
+     *
+     * @param resolvedTypes member types
+     * @return a 'BUnionType' instance by combining all its member types
+     */
+    public Value getUnionTypeFrom(List<Value> resolvedTypes) throws EvaluationException {
+        List<String> methodArgTypeNames = new ArrayList<>();
+        methodArgTypeNames.add(B_TYPE_ARRAY_CLASS);
+        RuntimeStaticMethod method = getRuntimeMethod(context, B_TYPE_CREATOR_CLASS, CREATE_UNION_TYPE_METHOD,
+                methodArgTypeNames);
+        List<Value> methodArgs = new ArrayList<>(resolvedTypes);
+        method.setArgValues(methodArgs);
+        return method.invokeSafely();
+    }
+
+    protected Optional<Value> resolveInbuiltType(String typeName) {
+        try {
+            List<String> methodArgTypeNames = Collections.singletonList(JAVA_STRING_CLASS);
+            RuntimeStaticMethod method = getRuntimeMethod(context, B_TYPE_UTILS_CLASS, VALUE_FROM_STRING_METHOD,
+                    methodArgTypeNames);
+            method.setArgValues(Collections.singletonList(context.getAttachedVm().mirrorOf(typeName)));
+            return Optional.of(new BExpressionValue(context, method.invokeSafely()).getJdiValue());
+        } catch (Exception e) {
+            return Optional.empty();
+        }
+    }
+
+    protected Optional<Symbol> getModuleTypeDefinitionSymbol(String typeName) {
+        SemanticModel semanticContext = context.getDebugCompiler().getSemanticInfo();
+        return semanticContext.moduleSymbols().stream()
+                .filter(symbol -> symbol.kind() == SymbolKind.TYPE_DEFINITION || symbol.kind() == SymbolKind.CLASS)
+                .filter(symbol -> encodeIdentifier(symbol.getName().get(), IdentifierModifier.IdentifierType.OTHER)
+                        .equals(typeName))
+                .findAny();
+    }
+
+    protected Optional<Symbol> getQualifiedTypeDefinitionSymbol(String modulePrefix, String typeName) {
+        return this.resolvedImports.get(modulePrefix).allSymbols().stream()
+                .filter(symbol -> symbol.kind() == SymbolKind.TYPE_DEFINITION || symbol.kind() == SymbolKind.CLASS)
+                .filter(symbol -> encodeIdentifier(symbol.getName().get(), IdentifierModifier.IdentifierType.OTHER)
+                        .equals(typeName))
+                .findAny();
+    }
+
+    protected String constructInitClassNameFrom(Symbol typeSymbol) {
+        ModuleID moduleMeta = typeSymbol.getModule().get().id();
+        return new StringJoiner(".")
+                .add(encodeModuleName(moduleMeta.orgName()))
+                .add(encodeModuleName(moduleMeta.moduleName()))
+                .add(moduleMeta.version().split("\\.")[0])
```

Comment:
Shall we move this hard coded string (`\\.`) to a constant? Do the similar changes in other places as well.